### PR TITLE
Fix UIConfig duplicate streamId bug + Tests

### LINKED ISSUE #628: UIConfig should check if stream belong to correct group of the process
When a stream exists on 2 groups in the ui config but only lives on one of theme on the process we display the stream only on the group it lives on in the process

### PR DESCRIPTION
**Pull request by O-Susanoo**
closes #628 

ironMan

I see that REACTor is all grown up now i wonder how many software years i'have been gone

### CODE REVIEW

File: src/utils/UIConfig/UIConfig.js [hunk 1]
```diff
@@ -48,4 +49,17 @@ export default class UIConfig {
     GraphsUIConfig.moveStream(graphs, groupName, sourceIndex, destinationIndex)
     return config
   }
+
+  static isGroupEmptyOnActiveProcess(group, activeProcess) {
+    const emptyGroup = group.streams
+      .map(stream => Process.getProcessStreamById(activeProcess, stream.streamId))
+      .every(stream => stream === undefined || stream.groupName !== group.name)
+    return emptyGroup
```
Comment (vdiezel) [inline src/utils/UIConfig/UIConfig.js]: your could immediately return here ` return group.streams.map....` :grimacing:

Comment (O-Susanoo) [reply src/utils/UIConfig/UIConfig.js]: copy paste is a disease that destroy lifes each year

Comment (O-Susanoo) [reply src/utils/UIConfig/UIConfig.js]: i will fix it

File: test/components/processviewer/body/dataviewer/DataGraphs.test.jsx [hunk 2]
```diff
@@ -340,4 +340,181 @@ describe('DataGraphs.jsx', () => {
...
+        groupStreams.forEach(streamName => {
+          const { getByText: getStreamByName } = within(group)
+          getStreamByName(streamName)
+        })
+      })
+
+      //check if stream in old group is not visible
+      const heatingGroup = getByText(mockedDevice.uiconfig.graphs.groups[0].name).parentNode.parentNode
+      const { queryByText: getStreamByName } = within(heatingGroup)
+      expect(getStreamByName(streamsWithSameGroup[0].name)).toBeNull()
```
Comment (vdiezel) [inline test/components/processviewer/body/dataviewer/DataGraphs.test.jsx]: :ok_hand:

File: src/utils/UIConfig/UIConfig.js (hunk 1 above)
Comment (vdiezel) [reply src/utils/UIConfig/UIConfig.js]: you don't have, too - just trying to make you aware of it in the future

**Pull request merged**